File: evergreen/2022/convert.js
const fs = require("fs");
var songsUnpruned = JSON.parse(fs.readFileSync("songs-unpruned.json"));
var positions = [];
songsUnpruned.data.cms_chart_editions.data[0].positions.forEach((song) => {
  position = {
    edition: "2022",
    id: parseInt(song.position),
    title: song.title,
    artist: song.artist,
    year: song.year,
    previous: parseInt(song.last_position),
  };
  positions.push(position);
});
fs.writeFileSync("songs.json", JSON.stringify(positions));

File: top2020/convert.py
#!/usr/bin/python3

import json
import requests

with open('songs-unpruned.json') as f:
    songs = json.load(f)

output = []
for song in songs['positions']:
    item = {'title': song['title'],
            'artist': song['artist'],
            'id': song['position'],
            'previous': song['lastPosition'],
            'image_id': song['imageUrl']}
    output.append(item)

with open('songs.json', 'w') as f:
    json.dump(output, f)

